source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst

```diff
@@ -1115,6 +1125,22 @@ clear the pool again. This situation is possible if the pool is cleared by the
 background thread after it encounters an error establishing a connection, but
 the ServerDescription for the endpoint was not updated accordingly yet.
 
+Why do we need force running prune maintenance call in Clear logic?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+If a SDAM logic has noticed a network error, we assume that all connections including
+"in use" connections are no longer healthy. In some cases we can't rely on throwing
+exception in the connection itself, for example, after network error, server request
+can hang on OS level in TCP retry loop and wait up to 17 minutes until failing.
+So, to notify pool about changes on network level asap, a pool forces running 
+maintenance logic to ensure that connection state is up to date. This logic is 
+cross driver and not dependent on OS behavior and configured maintenance interval.
+
+Why don't we configure TCP_USER_TIMEOUT?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
minor: `~~~~~~~` length should be same as previous line.

Reply:
I didn't realize this, done